File: game/calc/calc.c
#include <stdio.h>
#include <stdlib.h>

int main (int argc, char const* argv[])
{
    int a, b, err, ans;
    char op;

    scanf("%d %c %d", &a, &op, &b);

    err = 0;
    if (op == '+') {
        ans = a + b;
    } else if (op == '-') {
        ans = a - b;
    } else if (op == '*') {
        ans = a * b;
    } else if (op == '/') {
        ans = a / b;
    } else {
        err = 1;
    }

    if (err) {
        printf("An Error Occurred\n");
    } else {
        printf("%d\n", ans);
    }

    return 0;
}

File: game/calc/calc.pas
program calc;
var
    a, b, err, ans: integer;
    op: char;
begin
    read(a);
    writeln(a);
    read(op);
    writeln(op);
    read(b);
    writeln(b);

    err := 0;
    if (op = '+') then ans := a + b
    else if (op = '-') then ans := a - b
    else if (op = '*') then ans := a * b
    else if (op = '/') then ans := a div b
    else err := 1;

    if (err = 1) then writeln('An Error Occurred')
    else writeln(ans);
end.
